client/my-sites/email/inbox/index.js

```diff
@@ -33,26 +36,38 @@ const InboxManagement = ( { domains } ) => {
 		);
 	}
 
-	const domainsWithSubscriptions = domains.filter(
-		( domain ) => ! domain.isWPCOMDomain && hasPaidEmailWithUs( domain )
+	const nonStagingDomains = domains.filter( ( domain ) => ! domain.isWPCOMDomain );
+
+	// Find out if the site has at least one mailbox in the site
+	const hasAnyMailboxes = domains.some(
+		( domain ) =>
+			( nonStagingDomains &&
+				// it may be worth also checking for has(EmailProduct)WithUs( domain ) for each branch below
+				getGSuiteMailboxCount( domain ) > 0 ) ||
+			getConfiguredTitanMailboxCount( domain ) > 0
 	);
 
-	//EmailManagementHome logic will handle the case where there is only one domain to show directly the email comparison
-	//and also if there is no domain so, it will show a CTA to buy domain
-	if ( domainsWithSubscriptions.length === 0 ) {
-		return (
-			<CalypsoShoppingCartProvider>
-				<EmailManagementHome
-					emailListInactiveHeader={ getMainHeader() }
-					sectionHeaderLabel={ translate( 'Domains' ) }
-					showActiveDomainList={ false }
-				/>
-			</CalypsoShoppingCartProvider>
-		);
+	// Load the Inbox UI if at least one mailbox occurs
+	if ( hasAnyMailboxes ) {
+		return <MailboxSelectionList />;
 	}
 
-	//If we are at this point it means that we've at least have one subscription to show in mailbox selector
-	return <MailboxSelectionList />;
+	const domainsWithEmailServices = domains.filter(
+		( domain ) =>
+			nonStagingDomains && ( hasPaidEmailWithUs( domain ) || hasEmailForwards( domain ) )
+	);
+
+	const showActiveDomainList = domainsWithEmailServices.length > 0;
```

Comment:
Looking at this I think we should negate the meaning of this variable `showActiveDomainList` name to:

```suggestion
	const showActiveDomainList = domainsWithEmailServices.length <= 0;
```
That way we can fit the variable meaning to the prop name:

```js
<EmailManagementHome showActiveDomainList= { showActiveDomainList } .../>
```